quadsim/robots/motor_model.py

```diff
@@ -0,0 +1,104 @@
+"""Implements a simple DC motor model."""
+import enum
+import numpy as np
+from typing import Sequence, Union, Optional
+
+POSITION_INDEX = 0
+KP_INDEX = 1
+VELOCITY_INDEX = 2
+KD_INDEX = 3
+TORQUE_INDEX = 4
+COMMAND_DIMENSION = 5
+
+
+class MotorControlMode(enum.Enum):
+    """Different motor control modes.
+    See the documentation of MotorModel class for details."""
+
+    POSITION = 0
+    TORQUE = 1
+    HYBRID = 2
+
+
+class MotorModel:
+    """Implements a simple DC motor model for simulation.
+
+    To accurately model the motor behaviors, this class converts all motor
+    commands into torques, which could be send directly to the simulator.
+    Right now, 3 motor control modes are supported:
+    - POSITION: performs joint-level PD control
+    - TORQUE: directly takes in motor torque command
+    - HYBRID: takes in a 5-d tuple (pos, kp, vel, kd, torque), and output
+      torque is a sum of PD torque and additional torque.
+    """
+
+    def __init__(self, config) -> None:
+        """Initializes the class."""
+        self.config = config
+        self._motor_control_mode = self.config.motor_control_mode
+        self._num_motors = self.config.num_motors
+        self._kp = self.config.kps
+        self._kd = self.config.kds
+        self._strength_ratios = np.ones(self._num_motors)
+        # Thresholds
+        self._max_torque = self.config.max_torque
+        self._min_torque = self.config.min_torque
+        self._max_velocity = self.config.max_velocity
+        self._min_velocity = self.config.min_velocity
+        self._max_position = self.config.max_position
+        self._min_position = self.config.min_position
+
+    def set_motor_gains(
```

Comment:
is there any reason we got rid of the rest of the getters and setters but kept this one? Why must this be done manually through a function call rather than just happens in `__init__`?

Reply:
The hope is to allow dynamic change of motor gains / strength ratios on-the-fly. Say in a domain-randomization setting, you change the motor strength at the beginning of every episode, this would allow you to set the strengths without having to re-construct motormodel objects.

Reply:
Ok I definitely agree with you on having the ability to dynamically adjust gains, etc. on the fly.

Just a little confused by mixing the two code styles I guess:
1. using explicit getters and setters (which is not really pythonic afaik)
2. using `@property` to provide soft get and set functionality rather than always accessing object attributes directly

Reply:
Done. I'm using the python way of getters and setters now.